### PR TITLE
Remove a  prefix `gen4312` from a hostname

### PR DESCRIPTION
**Pull request by allburov**
The exporter are using original `hostname` value from celery's events data. 
Celery uses [anon_nodename](https://docs.celeryq.dev/en/stable/internals/reference/celery.utils.nodenames.html?highlight=anon_nodename#celery.utils.nodenames.anon_nodename) to get the value and the value has format "prefix@hostname".

The prefix part is different depending on type of a process:
1. For workers the prefix is a worker's name, e.g., `worker1@allburov`
2. For senders the prefix has a format `genPID@hostname` where PID is Process ID, e.g., `gen1231234@allburov`

The part with PID has high cardinality and it leads to too much metrics in Prometheus (as described in  https://github.com/danihodovic/celery-exporter/issues/116)

[Prometheus suggests](https://prometheus.io/docs/practices/naming/):
>CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.

----

This PR removes prefix with PID for task-sent events - it's the only event that usually has PID in the hostname. 

I decided to keep using the prefix for workers because one can run few workers on the same host by using `--hostname` flag and it'll help them to understand what is going on there.
```
$ celery -A proj worker --loglevel=INFO --concurrency=10 -n worker1@%h
$ celery -A proj worker --loglevel=INFO --concurrency=10 -n worker2@%h
$ celery -A proj worker --loglevel=INFO --concurrency=10 -n worker3@%h
```

I thought about doing it a bit different way - we could always set `hostname` label to the pure-hostname value (without a prefix) and add `worker` label for these types of events that we can say "it belongs to THIS worker", so we'll have three dimensions labels for most of metrics - `name, hostname, [worker]`

What are you thoughts, does it make sense? 
Give me a feedback and we can work on that in the right direction!

### CODE REVIEW

**Comment by danihodovic** [comment]
Good point, I'll give this a test run when I have some free time. Reviewing other PR's at the moment.

File: src/exporter.py [hunk 1]
```diff
@@ -103,6 +104,13 @@ def track_task_event(self, event):
 
         labels = {"name": task.name, "hostname": task.hostname}
 
+        if event['type'] == "task-sent":
+            # Remove the prefix which contains Process ID from hostname: gen5321@hostname => hostname
+            # Prometheus suggests it:
+            # "Do not use labels to store dimensions with high cardinality (many different label values)"
+            _, node = nodesplit(event['hostname'])
+            labels['hostname'] = node
```
Comment (danihodovic) [inline src/exporter.py:112]: Shouldn't we do this for metrics of all types?

```
celery_task_received_total{hostname="gen1@django-wtf-beat-7798d75d75-b9tk8",name="django_apps.core.github_tasks.index_contributors"} 0.0

celery_task_sent_total{hostname="celery@django-wtf-worker-0",name="django_apps.core.github_tasks.index_contributors"} 0.0

celery_task_failed_total{exception="",hostname="gen1@django-wtf-beat-7798d75d75-b9tk8",name="celery.backend_cleanup"} 0.0
```

Comment (allburov) [reply src/exporter.py:112]: You're right, didn't notice it

Comment (allburov) [reply src/exporter.py]: @danihodovic have another look please when you have time, I've cleaned up all `hostname` field

**Comment by allburov** [comment]
Sorry for a lot of force pushes, linters didn't like my code :)

**Comment by danihodovic** [comment]
Seems to work in my environment. Спасибо

**Comment by danihodovic** [comment]
Released as danihodovic/celery-exporter:0.5.2 docker image.

**Pull request merged**